### PR TITLE
added solutions for strings

### PR DESCRIPTION
**Pull request by wrackspurt**

### CODE REVIEW

File: strings/designer.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Output the design pattern.
+"""
+
+
+if __name__ == '__main__':
+    n, m = map(int, input().split())
+    for i in range(int(n/2)):
+        print(('.|.' * i).rjust(int((m - 2) / 2), '-') + '.|.' + ('.|.' * i).ljust(int((m - 2) / 2), '-'))
+    print('WELCOME'.center(m, '-'))
+    for i in reversed(range(int(n/2))):
```
Comment (mishindmitry) [inline strings/designer.py:26]: range has builtin mechanism to iterate in reverse way
https://docs.python.org/3/library/stdtypes.html#range

File: strings/find.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def count_substring(string, sub_string):
+    count = 0
+    pos = string.find(sub_string)
+    for _ in range(0, len(string)):
+        while pos != -1:
+            pos = string.find(sub_string, pos + 1)
+            count += 1
+    return count
```
Comment (mishindmitry) [inline strings/find.py]: If line12 returned `-1` which means no matches - why would we ever iterate for `len(string)` times in line 13? You can quit before it with zero matches

File: strings/find.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+if __name__ == '__main__':
+    string = input().strip()
+    sub_string = input().strip()
```
Comment (mishindmitry) [inline strings/find.py]: I didn't find a requirement that strings should be stripped. 
What if I want to count the amount of whitespaces in the given string?

File: strings/text_wrap.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+You are given a string S and width w.
+Your task is to wrap the string into a paragraph of width w.
+"""
+
+
+import textwrap
+
+
+def wrap(string, max_width):
+    return textwrap.TextWrapper(width=max_width).fill(text=string)
```
Comment (mishindmitry) [inline strings/text_wrap.py]: can you implement it without using `textwrap`?

**Review by mishindmitry — CHANGES_REQUESTED** [review]

File: strings/text_wrap.py (hunk 4 above)
Comment (wrackspurt) [reply strings/text_wrap.py]: Yes:
1.
```
#solution 2
def wrap(string, max_width):
    l = []
    for i in range(0, len(string), max_width):
        l.append(string[i:i + max_width])
    return '\n'.join(l)


if __name__ == '__main__':
    string, max_width = input(), int(input())
    result = wrap(string, max_width)
    print(result)
```
2.
```
# solution 3
def wrap(string, max_width):
    return '\n'.join([string[i:i + max_width] for i in range(0, len(string), max_width)])


if __name__ == '__main__':
    string, max_width = input(), int(input())
    result = wrap(string, max_width)
    print(result)
```

File: strings/find.py (hunk 3 above)
Comment (wrackspurt) [reply strings/find.py]: strip() was used in a pattern of submission on hackerrank and I couldn't change it when I added my solution there

File: strings/alphabetRangoli.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+if __name__ == '__main__':
+    try:
+        n = int(input('please, enter the number: '))
+        if n < 0:
+            print('the number is not whole!')
+        elif n == 0:
+            print("it's a zero!")
+        elif n >= 27:
+            print('the number is too big!')
```
Comment (mishindmitry) [inline strings/alphabetRangoli.py]: no need for such granular error handling, just `9 < n <27`

File: strings/capitalize.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+# Complete the solve function below.
+def solve(s):
+    """l = []
+    for i in s.split():
+        l.append(i.capitalize())
+    return ' '.join(l)"""
+    # return ' '.join([i.capitalize() for i in s])
+    for i in s.split():
+        s = s.replace(i, i.capitalize())
+    return s
```
Comment (mishindmitry) [inline strings/capitalize.py]: what happens for the string `asd asdasd 123asd` ?

File: strings/text_wrap.py (hunk 4 above)
Comment (mishindmitry) [reply strings/text_wrap.py]: can you please replace it in the code, not just in the comment?

File: strings/merge_the_tools.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Output Format:
+Print n/k lines where each line i contains string ui.
+
+"""
+
+
+def merge_the_tools(string, k):
+    n = len(string)
+    if n == k * k:
```
Comment (mishindmitry) [inline strings/merge_the_tools.py]: it seems the condition is not what is requested in the task. `k is a factor of n` means n can be divided by k evenly(without a residual), e.g. 2 is a factor of 10. In your code it seems the condition is as `k is a square root of n`

File: strings/minion.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+Note :
+Vowels are only defined as AEIOU. In this problem, Y is not considered a vowel.
+"""
+
+
+def minion_game(string):
+    vowels = 'AEIOU'
+    kevin = sum(len(string) - i for i in range(len(string)) if string[i] in vowels)
+    stuart = sum(len(string) - i for i in range(len(string)) if string[i] not in vowels)
```
Comment (mishindmitry) [inline strings/minion.py]: I think here's the wrong winner condition implemented.
Expected condition for Kevin: total amount of words which start with a vowel and are substrings of `string`.
Implemented condition for Kevin: summary of (some) substring lenghts which starts with a vowel.
E.g. for `BANANA`: 
expected list for Kevin is [`A`, `A`, `A`, `AN`, `AN`, `ANA`, `ANA`, `ANAN`, `ANANA`] and score is 9
implemented list for Kevin is [`ANANA`, `ANA`, `A`] and score is 9
It seems the results are the same at least for the hacker-rank test cases but I have a feeling something's wrong...

**Review by mishindmitry — CHANGES_REQUESTED** [review]
Hi, please take a look at the remarks below

File: strings/designer.py (hunk 1 above)
Comment (wrackspurt) [reply strings/designer.py:26]: I've read about it

File: strings/alphabetRangoli.py (hunk 5 above)
Comment (wrackspurt) [reply strings/alphabetRangoli.py]: ok, I'll rewrite this

File: strings/find.py (hunk 2 above)
Comment (wrackspurt) [reply strings/find.py]: I'll recast it

File: strings/capitalize.py (hunk 6 above)
Comment (wrackspurt) [reply strings/capitalize.py]: wrong output, I'll fix it

File: strings/merge_the_tools.py (hunk 7 above)
Comment (wrackspurt) [reply strings/merge_the_tools.py]: Oh, yes, you're right, I'll fix it

File: strings/text_wrap.py (hunk 4 above)
Comment (wrackspurt) [reply strings/text_wrap.py]: ok

File: strings/capitalize.py (hunk 6 above)
Comment (mishindmitry) [reply strings/capitalize.py]: still not quite fixed.
here's one more example `123asd aSd asdasd asd`
- only first char should capitalized
- there's no requirements to make chars in lowercase - charachters in uppercase should remain uppercase

**Review by mishindmitry — CHANGES_REQUESTED** [review]
Please take a look more carefully at `capitalize.py`

**Review by mishindmitry — APPROVED** [review]

**Pull request merged**